reference/conanfile/tools/microsoft.rst

```diff
@@ -632,6 +632,32 @@ Example:
         check_min_vs(self, "192")
 
 
+.. _conan_tools_microsoft_msvs_toolset:
+
+msvs_toolset()
+--------------
+
+Available since: `1.59.0 <https://github.com/conan-io/conan/releases/tag/1.59.0>`_
+
+.. code-block:: python
+
+    def msvs_toolset(conanfile)
+
+Returns the corresponding Visual Studio or msvs platform toolset based on the settings of the given ``conanfile``. For instance, it may return ``v143``
+for ``compiler=Visual Studio`` with ``compiler.version=17``. If ``compiler.toolset`` was set in settings, it has a priority and always returned.
```

Comment:
what does it return for other compilers?

Reply:
An empty string for any other cases. I just added it to the description now. Good catch.